node/test/Test/Pos/Client/Txp/Mode.hs

```diff
@@ -0,0 +1,291 @@
+    dbIterSource = DB.dbIterSourcePureDefault
+
+instance DB.MonadDB TxpTestInitMode where
+    dbPut = DB.dbPutPureDefault
+    dbWriteBatch = DB.dbWriteBatchPureDefault
+    dbDelete = DB.dbDeletePureDefault
+
+instance
+    (HasCoreConstants, SscHelpersClass ssc) =>
+    DB.MonadBlockDBGeneric (BlockHeader ssc) (Block ssc) Undo TxpTestInitMode
+  where
+    dbGetBlock  = DB.dbGetBlockPureDefault @ssc
+    dbGetUndo   = DB.dbGetUndoPureDefault @ssc
+    dbGetHeader = DB.dbGetHeaderPureDefault @ssc
+
+instance (HasCoreConstants, SscHelpersClass ssc) =>
+         DB.MonadBlockDBGenericWrite (BlockHeader ssc) (Block ssc) Undo TxpTestInitMode where
+    dbPutBlund = DB.dbPutBlundPureDefault
+
+instance
+    (HasCoreConstants, SscHelpersClass ssc) =>
+    DB.MonadBlockDBGeneric (Some IsHeader) (SscBlock ssc) () TxpTestInitMode
+  where
+    dbGetBlock  = DB.dbGetBlockSscPureDefault @ssc
+    dbGetUndo   = DB.dbGetUndoSscPureDefault @ssc
+    dbGetHeader = DB.dbGetHeaderSscPureDefault @ssc
+
+----------------------------------------------------------------------------
+-- Boilerplate TxpTestContext instances
+----------------------------------------------------------------------------
+
+instance GS.HasGStateContext TxpTestContext where
+    gStateContext = ttcGState_L
+
+instance HasSlottingVar TxpTestContext where
+    slottingTimestamp = ttcSystemStart_L
+    slottingVar = GS.gStateContext . GS.gscSlottingVar
+
+instance HasLens GenesisWStakeholders TxpTestContext GenesisWStakeholders where
+    lensOf = ttcParams_L . ttpGenesisContext . gtcWStakeholders
+
+instance HasLens DB.DBPureVar TxpTestContext DB.DBPureVar where
+    lensOf = GS.gStateContext . GS.gscDB . pureDBLens
+      where
+        getter = \case
+            DB.RealDB _   -> realDBInTestsError
+            DB.PureDB pdb -> pdb
+        setter _ pdb = DB.PureDB pdb
+        pureDBLens = lens getter setter
+        realDBInTestsError = error "You are using real db in tests"
+
+----------------------------------------------------------------------------
+-- Boilerplate TxpTestMode instances
+----------------------------------------------------------------------------
+
+instance MonadGState TxpTestMode where
+    gsAdoptedBVData = DB.gsAdoptedBVDataDefault
+
+instance (HasCoreConstants, MonadSlotsData ctx TxpTestMode)
+      => MonadSlots ctx TxpTestMode
+  where
+    getCurrentSlot = do
+        view ttcSlotId_L >>= \case
+            Nothing -> Slot.getCurrentSlotSimple =<< view ttcSSlottingVar_L
+            Just slot -> pure (Just slot)
+    getCurrentSlotBlocking =
+        view ttcSlotId_L >>= \case
+            Nothing -> Slot.getCurrentSlotBlockingSimple =<< view ttcSSlottingVar_L
+            Just slot -> pure slot
+    getCurrentSlotInaccurate = do
+        view ttcSlotId_L >>= \case
+            Nothing -> Slot.getCurrentSlotInaccurateSimple =<< view ttcSSlottingVar_L
+            Just slot -> pure slot
+    -- FIXME: this has already caused a bug with Explorer (CSE-203).
```

Comment:
Note that CSE-203 was fixed and `currentTimeSlottingSimple` should be ok now. I don't think FIXME still makes sense here.

Reply:
The fix for CSE-203 was exactly to replace `currentTimeSlottingSimple` with `Timestamp 0`.

Reply:
The fix for CSE-203 was not to use current time as timestamp in block processing, because it doesn't make sense.